osquery/main/windows/main.cpp

```diff
@@ -384,30 +323,20 @@ void WINAPI ServiceMain(DWORD argc, LPSTR* argv) {
     kServiceStatus.dwServiceType = SERVICE_WIN32_OWN_PROCESS;
     kServiceStatus.dwServiceSpecificExitCode = 0;
     UpdateServiceStatus(0, SERVICE_START_PENDING, 0, 0);
+    UpdateServiceStatus(
+        SERVICE_ACCEPT_STOP | SERVICE_ACCEPT_SHUTDOWN, SERVICE_RUNNING, 0, 0);
 
-    auto stopEvent =
```

Comment:
Have you tested this interacting with a watcher as a system service on Windows? I'll pull it down and check, but I'm a little concerned that we've moved away from events for synchronization between the two processes. I definitely could be wrong or just confused here as it's been a bit since I've worked around this code, but I'm concerned you're not going to be able to signal from the worker/watcher to the other process that a shutdown is occurring, and that one of the two processes will experience a non-graceful shutdown. 

I see above the shutdown logic looks like we're attempting to open up a handle to the main thread with the `kLegacyThreadId`, but how are we intending to communicate to the worker processes that a shutdown is occuring?

Reply:
So I guess disregard, it looks like the service is at least behaving correctly on my host and restarting:
```
PS C:\Users\Nicholas\work\repos\osquery> stop-service osqueryd
PS C:\Users\Nicholas\work\repos\osquery> sc.exe query osqueryd

SERVICE_NAME: osqueryd
        TYPE               : 10  WIN32_OWN_PROCESS
        STATE              : 1  STOPPED
        WIN32_EXIT_CODE    : 0  (0x0)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x0
```

Reply:
Yeah I see that the stop event was added to accommodate for my poor programming w.r.t. the termination state machine. With these new fixups the stop event pattern is no longer needed. I know you guys put a lot of work into creating it, and making the service shutdown reliable, sorry for causing that overhead a while back.